src/engine/SimulatorConfig.ts

```diff
@@ -8,7 +8,7 @@ export interface SimulatorConfig {
 export interface WorldConfig {
   xLength: number;
   zLength: number;
-  walls?: IWallSpec[];
+  walls?: IPerimeterSpec;
```

Comment:
In order to not break any existing users of this, make the type `IPerimeterSpec | IWallSpec[]`

Reply:
Actually, let's add this as a separate optional property on `WorldConfig`: `perimeter?: IPerimeterSpec` and you can leave `walls` as `walls?: IWallSpec[]`

Then, while configuring the world, if we see `perimeter`, favor that over the `walls` configuration